source/slang/slang-options.cpp

```diff
@@ -509,6 +510,13 @@ struct OptionsParser
 
                     SLANG_RETURN_ON_FAIL(StateSerializeUtil::extractFilesToDirectory(reproName));
                 }
+                else if (argStr == "-module-name")
+                {
+                    String moduleName;
+                    SLANG_RETURN_ON_FAIL(tryReadCommandLineArgument(sink, arg, &argCursor, argEnd, moduleName));
+
+                    spSetDefaultModuleName(compileRequest, moduleName.getBuffer());
+                }
```

Comment:
Instead of the `spSetDefaultModuleName` API, it seems like the `-module-name` option should set state on the `RawTranslationUnit` type, similar to how the `-stage` option works with the `RawEntryPoint` type.

This would require some cleanup work after all the options have been parsed, again similar to what is done for the `-stage` option, and then we would go through and call something like `spSetTranslationUnitName(compileRequest, translationUnitIndex, name)`.

(A more invasive approach could try to avoid calling `spAddTranslationUnit` until after all the options are parsed, so that we have the name(s) available, but that would potentially cause a lot of churn in the options parsing since much of it is set up around statefully poking at the compile request as options are parsed.)